src/plugins/data_source/server/saved_objects/credential_saved_objects_client_wrapper.ts

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import _ from 'lodash';
+
+import {
+  HttpServiceStart,
+  SavedObjectsBulkCreateObject,
+  SavedObjectsBulkResponse,
+  SavedObjectsBulkUpdateObject,
+  SavedObjectsBulkUpdateOptions,
+  SavedObjectsBulkUpdateResponse,
+  SavedObjectsClientWrapperFactory,
+  SavedObjectsCreateOptions,
+  SavedObjectsUpdateOptions,
+  SavedObjectsUpdateResponse,
+} from 'opensearch-dashboards/server';
+
+import { SavedObjectsErrorHelpers } from '../../../../core/server';
+
+import { CryptographyClient } from '../cryptography';
+
+import { Credential } from '../../common';
+
+export class CredentialSavedObjectsClientWrapper {
+  public httpStart?: HttpServiceStart;
+
+  private _type: string = 'credential';
+  private _authType: Credential.SharedAuthType = 'shared';
+  private _noAuthType: Credential.NoAuthType = 'no_auth';
+  private _usernamePasswordType: Credential.UsernamePasswordType = 'username_password';
+
+  private _cryptographyClient: CryptographyClient;
+
+  constructor(cryptographyClient: CryptographyClient) {
+    this._cryptographyClient = cryptographyClient;
+  }
+
+  public wrapperFactory: SavedObjectsClientWrapperFactory = (wrapperOptions) => {
+    const createWithCredentialMaterialsContentEncryption = async <T = unknown>(
```

Comment:
how about just `createWithCredentialMaterialsEncryption`, and maybe come comment to help understand what CredentialMaterials can be?

Reply:
added more comments and updated the function names for readability